specs/Picker-API_spec.md

```diff
@@ -0,0 +1,121 @@
+for the functionality: File and folder picker, Multiple folder picker.
+
+- Windows.Storage.Pickers.FileOpenPicker (https://docs.microsoft.com/en-us/uwp/api/Windows.Storage.Pickers.FileOpenPicker)
+- Windows.Storage.Pickers.FileSavePicker (https://docs.microsoft.com/en-us/uwp/api/windows.storage.pickers.filesavepicker)
+- Windows.Storage.Pickers.FolderPicker (https://docs.microsoft.com/en-us/uwp/api/windows.storage.pickers.folderpicker) 
+
+# Description
+
+The goal of this API is to address the gaps in the existing File/Folder picker APIs. The API surface is similar to the existing
+Picker APIs in Windows SDK with additional methods to support File + Folder picker and multiple folder picker. The Project Reunion APIs will supercede the Windows APIs.
+
+# Examples
+## Show file and folder picker
+```c#
+    var picker = new StorageItemPicker();
+    picker.FiltersAndDescriptions.Insert("Images", "*.jpg, *.png");
+    picker.StartLocation = startLocation;
+    picker.PickerKinds = StorageItemPickerKinds.Folder | StorageItemPickerKinds.FileOpen;
+    // multi-select will be automatically set
+    IReadOnlyList<Windows.Storage.IStorageItem> storageItems = await picker.ShowPickerAsync();
+
+    if(0 < storageItems.Count)
+    {
+    }
+```
+## Show single folder picker
+```c#
+    var picker = new StorageItemPicker();
+    picker.PickerKinds = StorageItemPickerKinds.Folder;
+    IReadOnlyList<Windows.Storage.IStorageItem> folder = await picker.ShowPickerAsync();
+```
+## Show multi folder picker
+```c#
+    var picker = new StorageItemPicker();
+    picker.PickerKinds = StorageItemPickerKinds.Folder;
+    picker.MultiSelect = true;
+    IReadOnlyList<Windows.Storage.IStorageItem> folders = await picker.ShowPickerAsync();
+```
+## Show multiple file picker
+```c#
+    var picker = new StorageItemPicker();
+    picker.StartLocation = startLocation;
+    picker.MultuSelect = true;
+    picker.PickerKinds = StorageItemPickerKinds.FileOpen;
+    IReadOnlyList<Windows.Storage.IStorageItem> files = await picker.ShowPickerAsync();
+```
+## Show File Save picker
+```c#
+    var picker = new StorageItemPicker();
+    picker.PickerKinds = StorageItemPickerKinds.FileSave;
+    IReadOnlyList<Windows.Storage.IStorageItem> fileToSave = await picker.ShowPickerAsync();
+```
+
+# Remarks
+# API Notes
+# API Details
+```c#
+    namespace Microsoft.Reunion.Storage.Pickers
+    {
+        enum PickerViewMode
+        {
+            List = 0,
+            Details,
+            Tiles,
+            Content
+        };
+
+        enum StorageItemPickerKinds
+        {
+            FileOpen = 0x1,
+            Folder = 0x2,
+            FileSave = 0x4
+        };
+
+        runtimeclass StorageItemPicker
+        {
+            StorageItemPicker();
+
+            /// Properties
+            String StartLocation;
+            String CommitButtonText;
+            String CancelButtonText;
+            Windows.Foundation.Collections.IPropertySet FiltersAndDescriptions{ get; };
```

Comment:
Property sets are unordered. In what order are the items displayed in the picker?

Reply:
Why `IPropertySet`? Wouldn't it be an `IVector<FileFilterInfo>` where `FileFilterInfo` has the name and list of extensions? Even doing a formatted string like `"Images;*.png,*.jpg"` seems OK if slightly less object-ish.

Reply:
It will be in the order added. Do we need to enforce an order?

---
In reply to: [499729488](https://github.com/microsoft/ProjectReunion/pull/99#discussion_r499729488) [](ancestors = 499729488)